src/message/decodeRpc.ts

```diff
@@ -20,236 +20,205 @@ export const defaultDecodeRpcLimits: DecodeRPCLimits = {
 }
 
 /**
- * Copied code from src/message/rpc.cjs but with decode limits to prevent OOM attacks
+ * Copied code from src/message/rpc.ts but with decode limits to prevent OOM attacks
  */
-export function decodeRpc (bytes: Uint8Array, opts: DecodeRPCLimits): IRPC {
+export function decodeRpc (bytes: Uint8Array, opts: DecodeRPCLimits): RPC {
```

Comment:
I think protons runtime now can limit fields while decoding.
Might be worth looking into as a followup PR.

Reply:
I can give it a try thanks @wemeetagain 👍